### PR TITLE
parse 'status' and 'HR number', validations check

### PR DESCRIPTION
**Pull request by mixman**

### CODE REVIEW

File: personio-client/src/Personio/EmployeeStatus.hs [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+import Data.Aeson.Compat (Value (String), withText)
+import Futurice.Generics
+import Futurice.Prelude
+import Prelude ()
+import Data.Swagger               (NamedSchema (..))
+
+import qualified Data.Map  as Map
+import qualified Data.Text as T
+
+data Status
```
Comment (phadej) [inline personio-client/src/Personio/EmployeeStatus.hs]: We have to document how this is related to the `hireDate` and `contractEndDate`, and what to do when they don't agree (e.g. `status = Active` but `contractEndDate` passed OR `status = Inactive` but `hireDate` passed etc).

**Comment by phadej** [comment]
General TODO: refactor to use `Futurice.Enum` like https://github.com/futurice/haskell-mega-repo/blob/a0af7cd125c40ebc65f3ae10b646f38032859784/fum-carbon-app/src/Futurice/App/FUM/Types/Status.hs

This cannot be done right now, as `Futurice.Enum` is WIP in `fum-carbon-app` (will be moved to `futurice-prelude`, eventually, but later defining this kind of enum types will be simpler

File: personio-client/package.yaml [hunk 2]
```diff
@@ -17,8 +17,10 @@ ghc-options: -Wall
 dependencies:
   - aeson
   - aeson-compat
+  - ansi-pretty
```
Comment (phadej) [inline personio-client/package.yaml]: ?

Comment (mixman) [inline personio-client/package.yaml]: Would be useful to be available from Futurice.Prelude

File: personio-client/src/Personio/Types.hs [hunk 3]
```diff
@@ -196,6 +197,7 @@ instance FromJSON SupervisorId where
 newtype NamedAttribute = NamedAttribute { getName :: Maybe Text }
 
 instance FromJSON NamedAttribute where
+    parseJSON (Null) = pure (NamedAttribute Nothing)
     parseJSON v = case v of
```
Comment (phadej) [inline personio-client/src/Personio/Types.hs]: would be better to move into `case` below, so we don't pattern match on the argument twice (though GHC can optimise those, yet still).

File: personio-client/package.yaml (hunk 2 above)
Comment (phadej) [inline personio-client/package.yaml]: I mean, I don't see any uses of `ansi-pretty` in the source

File: personio-client/src/Personio/EmployeeStatus.hs [hunk 4]
```diff
@@ -0,0 +1,57 @@
+{-# LANGUAGE DataKinds             #-}
+{-# LANGUAGE MultiParamTypeClasses #-}
+{-# LANGUAGE OverloadedStrings     #-}
+{-# LANGUAGE TemplateHaskell       #-}
+{-# LANGUAGE TypeFamilies          #-}
+{-# LANGUAGE DeriveGeneric         #-}
+module Personio.EmployeeStatus (
```
Comment (phadej) [inline personio-client/src/Personio/EmployeeStatus.hs]: noticed just now, should it be `Personio.Types.EmployeeStatus`?

File: personio-client/tests/Tests.hs [hunk 5]
```diff
@@ -10,6 +10,8 @@ import Test.Tasty
 import Test.Tasty.HUnit
 import Test.Tasty.QuickCheck
 
+import Personio.EmployeeStatus (Status(..))
```
Comment (phadej) [inline personio-client/tests/Tests.hs]: Should we re-export that module from `Personio.Types` (which is re-exported from `Personio`), so we don't need this import.

**Pull request merged**